src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,41 +1,85 @@
+/* eslint-disable no-console */
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+import cn from 'classnames';
+import { useDispatch } from 'react-redux';
+import { currentTodoSlice } from '../../features/currentTodo';
+
+type Props = {
+  selectedTodo: Todo;
+};
+
+const defaultUser: User = {
+  id: 0,
+  name: 'Default Username',
+  email: 'Default Email',
+  phone: 'Default Phonenumber',
+};
+
+export const TodoModal: React.FC<Props> = ({ selectedTodo }) => {
+  const [user, setUser] = useState<User>(defaultUser);
+  const dispatch = useDispatch();
+
+  const selectedUserId = selectedTodo.userId;
+
+  useEffect(() => {
+    getUser(selectedUserId)
+      .then(setUser)
+      .catch(() => console.error('Error while trying to load user!'));
+  }, [selectedUserId]);
+
+  const isLoaded = selectedTodo && user !== defaultUser;
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
+      {!isLoaded ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              Todo #{selectedTodo.id}
+            </div>
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              onClick={() => dispatch(currentTodoSlice.actions.viewTodo(null))}
```

Comment:
Ensure that the `viewTodo` action in `currentTodoSlice` is correctly defined to handle setting the selected todo to null, which effectively closes the modal.